Command run improvements

akihikodaki opened the pull request:
See each commit for details

akihikodaki commented on lib/auxiliary/host_helper.rb:
```diff
@@ -1,20 +1,16 @@
-      file.close
-      file.unlink
+    def temp_file(&block)
+      File.open Dir.tmpdir, File::RDWR | File::TMPFILE, &block
```
Dropped this change.

akihikodaki commented on lib/auxiliary/cmd_run.rb:
```diff
@@ -0,0 +1,53 @@
+      @cmd = cmd.join(' ')
+      logger.info("Run command: #{@cmd}")
+      @logger = logger
+      @stdout = File.open(Dir.tmpdir, File::RDWR | File::TMPFILE)
```
Fixed with commit d57934dfa3b1f1f87856ac61cf4c2bb4e4a0c554.

akihikodaki commented on lib/auxiliary/cmd_run.rb:
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
```
Fixed with commit d57934dfa3b1f1f87856ac61cf4c2bb4e4a0c554.

kostyanf14 reviewed (changes requested):
@YanVugenfirer 
What do you think about the last commit 'Use QMP instead of HMP'? This is from VirtHCK and we can use HMP in the manual run of AutoHCK and It is more simple than QMP + JSON in this case.

YanVugenfirer commented:
> @YanVugenfirer
> What do you think about the last commit 'Use QMP instead of HMP'? This is from VirtHCK and we can use HMP in the manual run of AutoHCK and It is more simple than QMP + JSON in this case.

The question what for we want to use it? If it only to quit to reboot the machine, than human still can send JSON. Do we have other use cases?

kostyanf14 commented:
> > @YanVugenfirer
> > What do you think about the last commit 'Use QMP instead of HMP'? This is from VirtHCK and we can use HMP in the manual run of AutoHCK and It is more simple than QMP + JSON in this case.
> 
> The question what for we want to use it? If it only to quit to reboot the machine, than human still can send JSON. Do we have other use cases?

Currently, AutoHCK uses it only for quitting and rebooting.

YanVugenfirer commented:
> > > @YanVugenfirer
> > > What do you think about the last commit 'Use QMP instead of HMP'? This is from VirtHCK and we can use HMP in the manual run of AutoHCK and It is more simple than QMP + JSON in this case.
> > 
> > 
> > The question what for we want to use it? If it only to quit to reboot the machine, than human still can send JSON. Do we have other use cases?
> 
> Currently, AutoHCK uses it only for quitting and rebooting.

"Philosophically" I think QMP is the better approach when SW uses QEMU. So let's go with it.

akihikodaki commented:
5584521d5bfe8c5a143691ec4a2a4f0c6e361487: now rewinding files in `AutoHCK::CmdRun`; otherwise it does not show anything.
0fb73a2809cae9fb95355be9e9925934485dc696: converted the socket to anonymous (libvirt does the same)
145a4ed7fc28fbccf7079bb797fbebfb119d7ba3: I changed my mind and left `@qemu_thread` present. It is now used to wait for QEMU with timeout.

Regarding the change to use QMP instead of HMP, I'm not concerned about their complexities much. I just chose QMP because HMP is somewhat unreliable for me and often said "qemu-monitor not responding" (and made me waiting for tens of seconds during debugging abortion bugs!)
